src/errors/errors.ts

```diff
@@ -1,153 +1,209 @@
-/**
- * Base class for all errors thrown by the sdk
- */
-abstract class SdkError extends Error {
-  protected constructor(message: string) {
-    super(message);
-  }
+export enum MomentoErrorCode {
+  // Invalid argument passed to Momento client
+  INVALID_ARGUMENT_ERROR = 'INVALID_ARGUMENT_ERROR',
+  // Service returned an unknown response
+  UNKNOWN_SERVICE_ERROR = 'UNKNOWN_SERVICE_ERROR',
+  // Cache with specified name already exists
+  ALREADY_EXISTS_ERROR = 'ALREADY_EXISTS_ERROR',
+  // Cache with specified name doesn't exist
+  NOT_FOUND_ERROR = 'NOT_FOUND_ERROR',
+  // An unexpected error occurred while trying to fulfill the request
+  INTERNAL_SERVER_ERROR = 'INTERNAL_SERVER_ERROR',
+  // Insufficient permissions to perform operation
+  PERMISSION_ERROR = 'PERMISSION_ERROR',
+  // Invalid authentication credentials to connect to cache service
+  AUTHENTICATION_ERROR = 'AUTHENTICATION_ERROR',
+  // Request was cancelled by the server
+  CANCELLED_ERROR = 'CANCELLED_ERROR',
+  // Request rate exceeded the limits for the account
+  LIMIT_EXCEEDED_ERROR = 'LIMIT_EXCEEDED_ERROR',
+  // Request was invalid
+  BAD_REQUEST_ERROR = 'BAD_REQUEST_ERROR',
+  // Client's configured timeout was exceeded
+  TIMEOUT_ERROR = 'TIMEOUT_ERROR',
+  // Server was unable to handle the request
+  SERVER_UNAVAILABLE = 'SERVER_UNAVAILABLE',
+  // A client resource (most likely memory) was exhausted
+  CLIENT_RESOURCE_EXHAUSTED = 'CLIENT_RESOURCE_EXHAUSTED',
```

Comment:
What is the difference between `CLIENT_RESOURCE_EXHAUSTED` and `LIMIT_EXCEEDED_ERROR`?

Reply:
The former is something like your laptop running out of memory. The latter is something like you've exceeded your request rate limit.

Reply:
client resources are things like memory on the client side